download-basket/src/main/java/downloadbasket/helpers/DownloadServices.java

```diff
@@ -0,0 +1,277 @@
+				ostream.close();
+			}
+		}
+		return returnFileName;
+	}
+	
+	/**
+	 * Check at is zipfile valid.
+	 * @param file zip file
+	 * @return
+	 */
+	static boolean isValid(final File file) {
+	    ZipFile zipfile = null;
+	    
+	    try {
+	        zipfile = new ZipFile(file);
+	        return true;
+	    } catch (ZipException e) {
+	        return false;
+	    } catch (IOException e) {
+	        return false;
+	    } finally {
+	        try {
+	            if (zipfile != null) {
+	                zipfile.close();
+	                zipfile = null;
+	            }
+	        } catch (IOException e) {
+	        }
+	    }
+	}
+	
+	/***
+	 * Send error report to support email.
+	 * @param xmlRequest xml request
+	 * @param wfsUrl wfs url
+	 */
+	
+	/**
+	 * Send error report to support email.
+	 * @param errorDetails error report details
+	 */
+	private void sendErrorReportToEmail(ErrorReportDetails errorDetails) {
+		try {
+			String msg = "";			
+			msg += "<b>Tapahtui virhe WFS-latauksessa</b><br/><br/>WFS-url:<br/>" + errorDetails.getWfsUrl() + "<br/><br/>"+
+					"WFS-pyyntö ja GeoServerin vastaus liitetiedostona.<br/><br/><b>HUOM!</b> Tämä on järjestelmän generoima virheilmoitus, älä vastaa tähän viestiin.";			
+
+			
+			//Using Multipart because HtmlEmail doesn't handle attachments very well.
+			MultiPartEmail email = new MultiPartEmail();
+			MimeBodyPart messageBodyPart = new MimeBodyPart();  
+			messageBodyPart.setContent(msg, "text/html; charset=UTF-8");
+			MimeMultipart multipart = new MimeMultipart();  
+			multipart.addBodyPart(messageBodyPart);
+			
+			byte[] bytes = errorDetails.getXmlRequest().getBytes();            
+
+            DataSource dataSource = new ByteArrayDataSource(bytes, "application/xml");
+            MimeBodyPart bodyPart = new MimeBodyPart();
+            bodyPart.setDataHandler(new DataHandler(dataSource));
+            bodyPart.setFileName("wfs_request.xml");
+            multipart.addBodyPart(bodyPart);
+            
+            if(errorDetails.getErrorFileLocation() != null){
+            	  DataSource source = new FileDataSource(errorDetails.getErrorFileLocation());
+            	  MimeBodyPart part = new MimeBodyPart();
+            	  part.setDataHandler(new DataHandler(source));
+            	  part.setFileName("geoserver_wfs_response.xml");
+                  multipart.addBodyPart(part);
+            } 
+            
+			email.setSmtpPort(Integer.parseInt(PropertyUtil.getNecessary(("hsy.wfs.download.smtp.port"))));
+			email.setCharset("UTF-8");
+					   
+			email.setContent(multipart);  
+			email.setHostName(PropertyUtil.getNecessary("hsy.wfs.download.smtp.host"));
+			email.setFrom(PropertyUtil.getNecessary("hsy.wfs.download.email.from"));
+			email.setSubject(PropertyUtil.getNecessary("hsy.wfs.download.error.report.subject"));
+			email.addTo(PropertyUtil.getNecessary("hsy.wfs.download.error.report.support.email"));
+			email.send();
+		}
+		catch (Exception ex) {
+			ex.printStackTrace();
```

Comment:
Use logger instead. Also the calling method might like to know that the email wasn't sent?